cypress/cypress/e2e/EngageSphere_GUI.cy.js

```diff
@@ -0,0 +1,57 @@
+// Renderiza o cabeçalho com um h1 e alternância de tema
+describe('Teste do cabeçalho e alternância de tema', () => {
+    beforeEach(() => {
+      cy.visit('http://localhost:3000/');
+    });
+
+    it('Deve renderizar o cabeçalho com um h1', () => {
+      cy.get('h1').should('exist');
```

Comment:
Que tal algo como o seguinte?

`cy.contains('h1', 'EngageSphere').should('be.visible)`

Veja que não só estou garantindo que o `h1` possui o texto correto, como também que está visível.

Um elemento pode estar no DOM e não estar visível, portanto, é importante verificar sua visibilidade.